packages/fileeditor-extension/src/commands.ts

```diff
@@ -178,7 +178,7 @@ export namespace Commands {
   ): void {
     config = filterUserSettings({
       ...CodeEditor.defaultConfig,
-      ...(settings.get('editorConfig').composite as JSONObject)
+      ...JSON.parse(settings.raw).editorConfig
```

Comment:
With #10880, this should not be needed.

```suggestion
      ...(settings.get('editorConfig').composite as JSONObject)
```